src/main/java/com/redhat/devtools/intellij/knative/kn/ServiceDeserializer.java

```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.intellij.knative.kn;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdNodeBasedDeserializer;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ServiceDeserializer extends StdNodeBasedDeserializer<Service> {
+
+    public ServiceDeserializer() {
+        super(Service.class);
+    }
+
+    @Override
+    public Service convert(JsonNode root, DeserializationContext deserializationContext) {
+        String name = root.get("metadata").get("name").asText();
+        JsonNode jsonStatus = root.get("status");
+
+        ServiceStatus status = convertToStatus(jsonStatus);
+
+        return new Service(name, status);
+    }
+
+    private ServiceStatus convertToStatus(JsonNode statusNode) {
+        String url = statusNode.get("url").asText();
+        int observedGeneration = statusNode.get("observedGeneration").asInt();
+        String latestReadyRevisionName = statusNode.get("latestReadyRevisionName").asText();
+        String latestCreatedRevisionName = statusNode.get("latestCreatedRevisionName").asText();
+        String addressUrl = statusNode.get("address").get("url").asText();
+
+        List<ServiceTraffic> traffic = convertToTraffic(statusNode.get("traffic"));
+        List<ServiceCondition> conditions = convertToConditions(statusNode.get("conditions"));
+
+        return new ServiceStatus(url, observedGeneration, latestReadyRevisionName, latestCreatedRevisionName, addressUrl, traffic, conditions);
+    }
+
+    private List<ServiceTraffic> convertToTraffic(JsonNode trafficNode) {
+        if (trafficNode == null || !trafficNode.isArray()) {
+            return null;
```

Comment:
If an empty value (e.g Collections.emptyList()) cannot exist, better use that as default value. If we can avoid using null, we are safer in future against NPE.
The same is true for Strings